### PR TITLE
serial: stm32: Use clock information from device tree

### LINKED ISSUE #10448: serial/stm32: use dts extracted information to populate clock settings
"clocks" property is missing in stm32 uart nodes, which prevents using #define CONFIG_xxx for clock related code configuration in uart instances:

cf: https://github.com/zephyrproject-rtos/zephyr/blob/4db9731451226d1a8f75ae1c355d98cce626d143/drivers/serial/uart_stm32.c#L356

Rather than using the hardcoded #defines LL_APB1_GRP1_P_XXX and STM32_CLOCK_BUS_XXX,
instance configuration code, should rather use the equivalent coming from dts #define generation.

Cf example in drivers/can/stm32_can.c:

static const struct can_stm32_config can_stm32_cfg_1 = {
        [... ]
	.pclken = {
		.enr = CONFIG_CAN_1_CLOCK_BITS,
		.bus = CONFIG_CAN_1_CLOCK_BUS,
	},
	.config_irq = config_can_1_irq
};

This should be done in 3 steps:
-Populate "clocks" property in all serial nodes in stm32 dtsi files (might be partially done already)
-Populate the CONFIG_ <> generated defined in stm32 fixup files
-Update stm32 serial instantiation code to use CONFIG_ symbols and  take advantage of this to simplify declaration preprocessor macros

### PR DESCRIPTION
**Pull request by erwango**
Fixes #10448

### CODE REVIEW

**Comment by erwango** [comment]
recheck

**Comment by erwango** [comment]
qemu issue again

**Comment by erwango** [comment]
recheck

**Review by ydamigos — APPROVED** [review]
Looks, good. Please consider removing the extra lines and fix a typo in commit's message (instanciation -> instantiation)

File: drivers/serial/uart_stm32.c [hunk 1]
```diff
@@ -363,10 +363,7 @@ static int uart_stm32_init(struct device *dev)
 	return 0;
 }
 
-/* Define clocks */
-#define STM32_CLOCK_UART(clock_bus, clock_enr)	\
-	.pclken = { .bus = clock_bus,		\
-		    .enr = clock_enr }
+
```
[the remark this answers is not in the document]
Comment (erwango) [reply drivers/serial/uart_stm32.c]: @ydamigos , done (remove the one I added). Thanks for review.

**Pull request merged**